lib/dangermattic/plugins/common/common_release_checks.rb

```diff
@@ -0,0 +1,122 @@
+  # It can be used directly or via the specialised plugins `AndroidReleaseCheck` and `IosReleaseCheck`.
+  #
+  # @example Checking if a specific file has changed on a release branch:
+  #          common_release_checks.check_file_changed(
+  #            file_comparison: ->(path) { path == 'metadata/full_release_notes.txt' },
+  #            message: 'Release notes have been modified on a release branch.',
+  #            on_release: true
+  #          )
+  #
+  # @example Checking if release notes and store strings have changed:
+  #          common_release_checks.check_release_notes_and_store_strings(
+  #            release_notes_file: 'metadata/release_notes.txt',
+  #            po_file: 'metadata/PlayStoreStrings.po'
+  #          )
+  #
+  # @example Checking for changes in internal release notes:
+  #          common_release_checks.check_internal_release_notes_changed
+  #
+  # @see Automattic/dangermattic
+  # @tags util, process, release
+  #
+  class CommonReleaseChecks < Plugin
+    DEFAULT_INTERNAL_RELEASE_NOTES = 'RELEASE-NOTES.txt'
+
+    # Check if certain files have been modified, returning a warning or failure message based on the branch type.
+    #
+    # @param file_comparison [Proc] Function used to compare modified file paths.
+    #   It should take a single argument, which is the path to a modified file,
+    #   and return true if the file matches the desired condition.
+    #   Example: `file_comparison = ->(file_path) { file_path.include?('app/') }`
+    #
+    # @param message [String] The message to display in the warning or failure output if the condition is met.
+    #
+    # @param on_release [Boolean] If true, the check will only run on release branches, otherwise on non-release branches.
+    #
+    # @param fail_on_error [Boolean] If true, a failure message will be displayed instead of a warning.
+    #
+    # @example Check if any modified file is under the 'app/' directory and emit a warning on release branches:
+    #   check_file_changed(file_comparison: ->(file_path) { file_path.include?('app/') },
+    #                      message: 'Some files in the "app/" directory have been modified. Please review the changes.',
+    #                      on_release: true)
+    #
+    # @example Check if a specific file has been modified and emit a failure on non-release branches:
+    #   check_file_changed(file_comparison: ->(file_path) { file_path == 'path/to/file/DoNotChange.java' },
+    #                      message: 'The "DoNotChange.java" file has been modified. This change is not allowed on non-release branches.',
+    #                      on_release: false,
+    #                      fail_on_error: true)
+    #
+    # @return [void]
+    def check_file_changed(file_comparison:, message:, on_release:, fail_on_error: false)
+      has_modified_file = git_utils.all_changed_files.any?(&file_comparison)
+
+      should_be_changed = on_release ? release_branch? : !release_branch?
```

Comment:
Might be more readable if you wrote this boolean expression like this? 🤷 
```suggestion
      should_be_changed = (on_release == release_branch?)
```

Reply:
Yeah, I think it reads better. Changed here: 4b83ceabe6e73dec3a0d6de3c39e553e24704a1a